avm/res/dashboard/grafana/main.bicep

```diff
@@ -0,0 +1,184 @@
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param apiKey string = 'Disabled'
+
+@allowed([
+  'Essential'
+  'Standard'
+])
+@description('Specifies the sku of the Azure Managed Grafana resource.')
+param skuName string = 'Standard'
+
+@allowed([
+  'TenantReuse'
+])
+param autoGeneratedDomainNameLabelScope string = 'TenantReuse'
+param deterministicOutboundIP string = 'Disabled'
+param enterpriseConfigurations grafanaTypes.enterpriseConfigurationsType = {}
+param grafanaIntegrations grafanaTypes.grafanaIntegrationsType = {}
+
+param grafanaPlugins object = {}
+
+param grafanaMajorVersion string = '10'
+
+@description('Optional. Specifies the the state for enable or disable traffic over the public interface for the the Azure Managed Grafana resource.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param publicNetworkAccess string = 'Disabled'
+
+@description('Optional. The zone redundancy setting of the Azure Managed Grafana resource.')
+@allowed([
+  'Disabled'
+  'Enabled'
+])
+param zoneRedundancy string = 'Disabled'
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints grafanaTypes.privateEndpointType
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' =
+  if (enableTelemetry) {
+    name: '46d3xbcp.res.bashboard.grafana.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+    properties: {
+      mode: 'Incremental'
+      template: {
+        '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+        contentVersion: '1.0.0.0'
+        resources: []
+        outputs: {
+          telemetry: {
+            type: 'String'
+            value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+          }
+        }
+      }
+    }
+  }
+
+param grafanaConfigurations grafanaTypes.grafanaConfigurationsType = {}
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities grafanaTypes.managedIdentitiesType
+
+@description('Optional. The lock settings of the service.')
+param lock grafanaTypes.lockType
+
+var formattedUserAssignedIdentities = reduce(
+  map((managedIdentities.?userAssignedResourcesIds ?? []), (id) => { '${id}': {} }),
+  {},
+  (cur, next) => union(cur, next)
+) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities)
+  ? {
+      type: (managedIdentities.?systemAssigned ?? false)
+        ? (!empty(managedIdentities.?userAssignedResourcesIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned')
+        : (!empty(managedIdentities.?userAssignedResourcesIds ?? {}) ? 'UserAssigned' : '')
+      userAssignedIdentities: !empty(formattedUserAssignedIdentities) ? formattedUserAssignedIdentities : null
+    }
+  : null
+
+resource grafana 'Microsoft.Dashboard/grafana@2023-09-01' = {
```

Comment:
Module seems to be missing role assignment implementation that's supported by Grafana as seen here: https://learn.microsoft.com/en-us/azure/managed-grafana/quickstart-managed-grafana-portal#access-your-azure-managed-grafana-instance